packages/libs/sdk/src/solana/solana.ts

```diff
@@ -0,0 +1,168 @@
+      await this.createDynamicPriorityFeeInstruction(feeLevel);
+    const allInstructions = [
+      ...transaction.instructions,
+      computeUnitPriceInstruction,
+    ];
+
+    // eslint-disable-next-line prefer-const
+    let [units, recentBlockhash] = await Promise.all([
+      this.getSimulationUnits(
+        this.connection,
+        allInstructions,
+        payerPublicKey,
+        feeLevel
+      ),
+      this.connection.getLatestBlockhash(),
+    ]);
+
+    transaction.add(computeUnitPriceInstruction);
+    if (units) {
+      units = Math.ceil(units * 1.05); // margin of error
+      transaction.add(ComputeBudgetProgram.setComputeUnitLimit({ units }));
+    }
+    transaction.recentBlockhash = recentBlockhash.blockhash;
+
+    return transaction;
+  }
+
+  private async createDynamicPriorityFeeInstruction(
+    feeType: PriorityFeeLevels = 'medium'
+  ) {
+    const { result } = await this.fetchEstimatePriorityFees({});
+    const priorityFee = result.per_compute_unit[feeType];
+    const priorityFeeInstruction = ComputeBudgetProgram.setComputeUnitPrice({
+      microLamports: priorityFee,
+    });
+    return priorityFeeInstruction;
+  }
+
+  private async fetchEstimatePriorityFees({
+    last_n_blocks = 100,
+    account = undefined,
+  }: EstimatePriorityFeesParams): Promise<PriorityFeeResponseData> {
+    const params: { last_n_blocks?: number; account?: string } = {};
+    if (last_n_blocks !== undefined) {
+      params.last_n_blocks = last_n_blocks;
+    }
+    if (account !== undefined) {
+      params.account = account;
+    }
+
+    const payload: PriorityFeeRequestPayload = {
+      method: 'qn_estimatePriorityFees',
+      params,
+      id: 1,
+      jsonrpc: '2.0',
+    };
+
+    const response = await fetch(this.endpointUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    });
+
+    if (!response.ok) {
+      if (response.status === 404) {
+        throw new Error(
+          `The RPC method qn_estimatePriorityFees was not found on your endpoint! Your endpoint likely does not have the Priority Fee API add-on installed. Please visit https://marketplace.quicknode.com/add-on/solana-priority-fee to install the Priority Fee API and use this method to send your transactions with priority fees calculated with real-time data.`
+        );
+      }
+      throw new Error('Failed to fetch priority fee estimates');
+    }
+
+    const data: PriorityFeeResponseData = await response.json();
+    return data;
+  }
+
+  private async getSimulationUnits(
+    connection: Connection,
+    instructions: TransactionInstruction[],
+    publicKey: PublicKey,
+    feeLevel: PriorityFeeLevels
+  ): Promise<number | undefined> {
+    const computeUnitPriceInstruction =
+      await this.createDynamicPriorityFeeInstruction(feeLevel);
+    const testInstructions = [...instructions, computeUnitPriceInstruction];
```

Comment:
I think you don't need this since you pass `allInstructionsn` 
```suggestion
    const testInstructions = instructions, computeUnitPriceInstruction;
```